bestofjs/bestofjs
# Improve database `repos` schema

## Goal

In the `repos` table we store both the `full_name` of the repo and the `name`, which is redundant since the full name is made of the `owner` and the `name` and not convenient when we need to filter repos by owner.

Changes:

- add `owner` column: the migration file includes the script to initialize it (technically we rename the `full_name` column)
- remove `full_name`
- make `owner_id` a number instead of a string (since the GitHub REST API returns numbers)
- use `drizzle-kit migrate` command for migrations
- Use a transaction when creating projects from the admin app, to avoid creating a repo linked to no project if the project creation fails
- Add `fullName` and `slug` parameters to CLI tasks, to be able to target either one project by `slug` or one repo by `fullName`
- Avoid creating too many connections to the database and simplify connections by using the exported `db` object

## How to test

Ensure all apps run as expected, especially projects pages showing list of projects

Check the CLI tasks with `fullName` and `slug` parameters

```sh
bun run apps/backend/src/cli.ts hello-world-repos --slug best-of-js --logLevel 4

bun run apps/backend/src/cli.ts hello-world-repos --fullName bestofjs/bestofjs --logLevel 4 
```
